onnx/defs/nn/defs.cc

```diff
@@ -126,6 +126,9 @@ void convPoolShapeInference(
             residual -= stride;
           }
         }
+        if (i >= static_cast<int>(effective_kernel_shape.size())) {
+          fail_shape_inference("kernel shape should have ", input_dims_size, " values in ", ctx.getDisplayName(), ".");
```

Comment:
```suggestion
          fail_shape_inference("kernel shape should have ", input_dims_size, " values in ", ctx.getDisplayName(), ".");
```

I would include the actual value

Reply:
The message displays the kernel size. I think that's ok. Compare the existing error messages, this one is very verbose.